util/misc.go

```diff
@@ -531,3 +543,62 @@ func GetLocalIP() string {
 	}
 	return ""
 }
+
+func createTLSCertificates() error {
+
+	privkey, err := rsa.GenerateKey(rand.Reader, 4096)
+	if err != nil {
+		return err
+	}
+
+	notBefore := time.Now()
+	notAfter := notBefore.Add(90 * 24 * time.Hour)
+	hostname, err := os.Hostname()
+	if err != nil {
+		return err
+	}
+
+	template := x509.Certificate{
+		SerialNumber: big.NewInt(1),
+		NotBefore:    notBefore,
+		NotAfter:     notAfter,
+		DNSNames:     []string{hostname},
+	}
+	derBytes, err := x509.CreateCertificate(rand.Reader, &template, &template, &privkey.PublicKey, privkey)
+	if err != nil {
+		return err
+	}
+
+	tempStoragePath := config.GetGlobalConfig().TempStoragePath
```

Comment:
I think this can be given as arguments or output for `createTLSCertificates()` so they are always in sync with the caller (where it is currently created again).